CDC-Data-Reconciliation-Frontend/src/components/Report.jsx

```diff
@@ -110,100 +268,311 @@ export default function Report({ reportID }) {
+                    </button>
+                    <button
+                      className='border rounded p-1 border-slate-400'
+                      onClick={() => statTable.setPageIndex(statTable.getPageCount() - 1)}
+                      disabled={!statTable.getCanNextPage()}
+                    >
+                      <MdKeyboardDoubleArrowRight className='text-xl' />
+                    </button>
+                    <span className='flex items-center gap-1'>
+                      <div>Page</div>
+                      <strong>
+                        {statTable.getState().pagination.pageIndex + 1} of {statTable.getPageCount()}
+                      </strong>
+                    </span>
+                    <span className='flex items-center gap-1'>
+                      | Go to page:
+                      <input
+                        type='number'
+                        defaultValue={statTable.getState().pagination.pageIndex + 1}
+                        onChange={(e) => {
+                          const page = e.target.value ? Number(e.target.value) - 1 : 0
+                          statTable.setPageIndex(page)
+                        }}
+                        className='border p-1 rounded w-16 border-slate-400'
+                      />
+                    </span>
+                    <select
+                      className='border p-1 rounded border-slate-400'
+                      value={statTable.getState().pagination.pageSize}
+                      onChange={(e) => {
+                        statTable.setPageSize(Number(e.target.value))
+                      }}
+                    >
+                      {[5, 10, 20, 30, 40, 50, 100].map((pageSize) => (
+                        <option key={pageSize} value={pageSize}>
+                          Show {pageSize}
+                        </option>
+                      ))}
+                    </select>
+                  </div>
+                </div>
               )}
             </>
           )}
 
-          <div className='flex flex-col items-center mb-4'>
-            <button
-              type='button'
-              className='bg-blue-400 text-white px-5 py-2 rounded-md hover:bg-blue-600'
-              onClick={handleDownload}
-            >
-              Download CSV
-            </button>
+          <div>
+            <div className='w-full flex flex-row items-center justify-between'>
+              <DebouncedInput
+                value={discGlobalFilter ?? ""}
+                onChange={(value) => setDiscGlobalFilter(String(value))}
+                className='p-2 font-lg shadow border border-block'
+                placeholder='Search all columns...'
+              />
+              <button
+                type='button'
+                className='bg-blue-400 text-white px-5 py-2 rounded-md hover:bg-blue-600 flex flex-row items-center justify-around gap-2'
+                onClick={handleDownload}
```

Comment:
A bit of a nit: I think this function should be renamed to handleResultsDownload.